File: Easy/asset/meta_assetsegview/meta_assetsegview.cs

/*
Easy
Copyright (C) 2021 Università degli Studi di Catania (www.unict.it)
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/


using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_assetsegview
{
    public class Meta_assetsegview : Meta_easydata 
	{
        public Meta_assetsegview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "assetsegview") {
				Name = "Beni strumentali";
			EditTypes.Add("seg");
			ListingTypes.Add("seg");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idasset","idpiece"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "seg": {
						DescribeAColumn(T, "inventory_description", "Inventario", nPos++);
						DescribeAColumn(T, "idasset", "Identificativo", nPos++);
						DescribeAColumn(T, "idpiece", "Numero parte", nPos++);
						DescribeAColumn(T, "assetacquire_description", "Descrizione", nPos++);
						DescribeAColumn(T, "asset_ninventory", "Numero inventario", nPos++);
						DescribeAColumn(T, "asset_rfid", "Codice RFID", nPos++);
						DescribeAColumn(T, "asset_lifestart", "Data inizio esistenza", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/asset/meta_assetsegview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_assetsegview() {
        MetaData.apply(this, ["assetsegview"]);
        this.name = 'meta_assetsegview';
    }

    meta_assetsegview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_assetsegview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'seg':
						this.describeAColumn(table, 'inventory_description', 'Inventario', null, 10, 50);
						this.describeAColumn(table, 'asset_ninventory', 'Numero inventario', null, 20, null);
						this.describeAColumn(table, 'idasset', 'Identificativo', null, 30, null);
						this.describeAColumn(table, 'idpiece', 'Numero parte', null, 40, null);
						this.describeAColumn(table, 'assetacquire_description', 'Descrizione Descrizione', null, 50, 150);
						this.describeAColumn(table, 'asset_rfid', 'Codice RFID', null, 60, 30);
						this.describeAColumn(table, 'asset_lifestart', 'Data inizio esistenza', null, 70, null);
						this.describeAColumn(table, 'inventorytree_codeinv', 'Codice Class. Inv.', null, 20, 50);
						this.describeAColumn(table, 'inventorytree_description', 'Denominazione Class. Inv.', null, 20, 150);
						this.describeAColumn(table, 'inventorytree_idinv', 'Class. Inv. Class. Inv.', null, 20, null);
						this.describeAColumn(table, 'upb_codeupb', 'Codice UPB', null, 30, 50);
						this.describeAColumn(table, 'upb_title', 'Denominazione UPB', null, 30, 150);
						this.describeAColumn(table, 'upb_idupb', 'UPB UPB', null, 30, 36);
//$objCalcFieldConfig_seg$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idasset", "idpiece"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('assetsegview', new meta_assetsegview('assetsegview'));

	}());
